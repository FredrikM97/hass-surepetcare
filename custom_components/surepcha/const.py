""" "Constants for the Sure Petcare integration."""

DOMAIN = "surepcha"

CONF_EMAIL = "email"
CONF_PASSWORD = "password"

TOKEN = "token"
CLIENT_DEVICE_ID = "client_device_id"
FACTORY = "surepetcare_factory"
TIMEOUT_API = 10
KEY_API = "surepetcare_api"
COORDINATOR_LIST = "coordinator_list"
COORDINATOR_DICT = "coordinator_dict"
COORDINATOR = "coordinator"
ENTRY_ID = "entry_id"
SCAN_INTERVAL = 300
POLLING_SPEED = "polling_speed"
LOCATION_INSIDE = "location_inside"
LOCATION_OUTSIDE = "location_outside"
OPTION_DEVICES = "devices"
DEVICE_OPTION = "device_option"
OPTIONS_FINISHED = "finished"
PRODUCT_ID = "product_id"
DEVICES = "devices"
NAME = "name"
