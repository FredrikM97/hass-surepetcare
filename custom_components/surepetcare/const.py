"""TODO."""

DOMAIN = "surepetcare"

CONF_EMAIL = "email"
CONF_PASSWORD = "password"

TOKEN = "token"
CLIENT_DEVICE_ID = "client_device_id"
FACTORY = "surepetcare_factory"
TIMEOUT_API = 10
KEY_API = "surepetcare_api"
COORDINATOR_LIST = "coordinator_list"
COORDINATOR_DICT = "coordinator_dict"
COORDINATOR = "coordinator"
