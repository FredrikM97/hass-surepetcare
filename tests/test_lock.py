import pytest
from unittest.mock import patch
from syrupy.assertion import SnapshotAssertion

from homeassistant.const import Platform
from homeassistant.core import HomeAssistant
from homeassistant.helpers import entity_registry as er

from . import initialize_entry

from pytest_homeassistant_custom_component.common import (
    MockConfigEntry,
    snapshot_platform,
)


@patch("custom_components.surepetcare.PLATFORMS", [Platform.LOCK])
@pytest.mark.usefixtures("enable_custom_integrations")
@pytest.mark.usefixtures("entity_registry_enabled_default")
@pytest.mark.asyncio
async def test_platform_setup_and_discovery(
    hass: HomeAssistant,
    mock_client,
    mock_config_entry: MockConfigEntry,
    entity_registry: er.EntityRegistry,
    snapshot: SnapshotAssertion,
    mock_devices,
    mock_pets,
) -> None:
    await initialize_entry(
        hass, mock_client, mock_config_entry, mock_devices, mock_pets
    )
    await snapshot_platform(hass, entity_registry, snapshot, mock_config_entry.entry_id)


@patch("custom_components.surepetcare.PLATFORMS", [Platform.LOCK])
@pytest.mark.usefixtures("enable_custom_integrations")
@pytest.mark.usefixtures("entity_registry_enabled_default")
@pytest.mark.asyncio
async def test_platform_setup_and_discovery_missing_entities(
    hass: HomeAssistant,
    mock_client,
    mock_config_entry_missing_entities: MockConfigEntry,
    entity_registry: er.EntityRegistry,
    snapshot: SnapshotAssertion,
    mock_devices,
    mock_pets,
) -> None:
    await initialize_entry(
        hass, mock_client, mock_config_entry_missing_entities, mock_devices, mock_pets
    )
    await snapshot_platform(
        hass, entity_registry, snapshot, mock_config_entry_missing_entities.entry_id
    )

@patch("custom_components.surepetcare.PLATFORMS", [Platform.LOCK])
@pytest.mark.usefixtures("enable_custom_integrations")
@pytest.mark.usefixtures("entity_registry_enabled_default")
@pytest.mark.asyncio
async def test_lock_toggle_and_snapshot(
    hass,
    mock_client,
    mock_config_entry: MockConfigEntry,
    entity_registry: er.EntityRegistry,
    snapshot: SnapshotAssertion,
    mock_devices,
    mock_pets,
):
    await initialize_entry(
        hass, mock_client, mock_config_entry, mock_devices, mock_pets
    )
    await hass.async_block_till_done()

    for entity_id in hass.states.async_entity_ids("lock"):
        # Lock
        await hass.services.async_call(
            "lock",
            "lock",
            {"entity_id": entity_id},
            blocking=True,
        )
        await hass.async_block_till_done()
        state_locked = hass.states.get(entity_id)
        assert state_locked is not None
        snapshot(name=f"{entity_id}-locked").assert_match(state_locked)

        # Unlock
        await hass.services.async_call(
            "lock",
            "unlock",
            {"entity_id": entity_id},
            blocking=True,
        )
        await hass.async_block_till_done()
        state_unlocked = hass.states.get(entity_id)
        assert state_unlocked is not None
        snapshot(name=f"{entity_id}-unlocked").assert_match(state_unlocked)