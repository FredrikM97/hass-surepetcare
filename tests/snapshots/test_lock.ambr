# serializer version: 1
# name: test_lock_toggle_and_snapshot[lock.katzenklappe_locking-locked]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Katzenklappe Locking',
      'supported_features': <LockEntityFeature: 0>,
    }),
    'context': <ANY>,
    'entity_id': 'lock.katzenklappe_locking',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unlocked',
  })
# ---
# name: test_lock_toggle_and_snapshot[lock.katzenklappe_locking-unlocked]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Katzenklappe Locking',
      'supported_features': <LockEntityFeature: 0>,
    }),
    'context': <ANY>,
    'entity_id': 'lock.katzenklappe_locking',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unlocked',
  })
# ---
# name: test_lock_toggle_and_snapshot[lock.laundry_door_locking-locked]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Laundry Door Locking',
      'supported_features': <LockEntityFeature: 0>,
    }),
    'context': <ANY>,
    'entity_id': 'lock.laundry_door_locking',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unlocked',
  })
# ---
# name: test_lock_toggle_and_snapshot[lock.laundry_door_locking-unlocked]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Laundry Door Locking',
      'supported_features': <LockEntityFeature: 0>,
    }),
    'context': <ANY>,
    'entity_id': 'lock.laundry_door_locking',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unlocked',
  })
# ---
# name: test_platform_setup_and_discovery[lock.katzenklappe_locking-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'lock',
    'entity_category': None,
    'entity_id': 'lock.katzenklappe_locking',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Locking',
    'platform': 'surepcha',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'locking',
    'unique_id': '1299453-locking',
    'unit_of_measurement': None,
  })
# ---
# name: test_platform_setup_and_discovery[lock.katzenklappe_locking-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Katzenklappe Locking',
      'supported_features': <LockEntityFeature: 0>,
    }),
    'context': <ANY>,
    'entity_id': 'lock.katzenklappe_locking',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unlocked',
  })
# ---
# name: test_platform_setup_and_discovery[lock.laundry_door_locking-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'lock',
    'entity_category': None,
    'entity_id': 'lock.laundry_door_locking',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Locking',
    'platform': 'surepcha',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'locking',
    'unique_id': '727608-locking',
    'unit_of_measurement': None,
  })
# ---
# name: test_platform_setup_and_discovery[lock.laundry_door_locking-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Laundry Door Locking',
      'supported_features': <LockEntityFeature: 0>,
    }),
    'context': <ANY>,
    'entity_id': 'lock.laundry_door_locking',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unlocked',
  })
# ---
# name: test_platform_setup_and_discovery_missing_entities[lock.katzenklappe_locking-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'lock',
    'entity_category': None,
    'entity_id': 'lock.katzenklappe_locking',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Locking',
    'platform': 'surepcha',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'locking',
    'unique_id': '1299453-locking',
    'unit_of_measurement': None,
  })
# ---
# name: test_platform_setup_and_discovery_missing_entities[lock.katzenklappe_locking-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Katzenklappe Locking',
      'supported_features': <LockEntityFeature: 0>,
    }),
    'context': <ANY>,
    'entity_id': 'lock.katzenklappe_locking',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unlocked',
  })
# ---
# name: test_platform_setup_and_discovery_missing_entities[lock.laundry_door_locking-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'lock',
    'entity_category': None,
    'entity_id': 'lock.laundry_door_locking',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Locking',
    'platform': 'surepcha',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'locking',
    'unique_id': '727608-locking',
    'unit_of_measurement': None,
  })
# ---
# name: test_platform_setup_and_discovery_missing_entities[lock.laundry_door_locking-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Laundry Door Locking',
      'supported_features': <LockEntityFeature: 0>,
    }),
    'context': <ANY>,
    'entity_id': 'lock.laundry_door_locking',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unlocked',
  })
# ---
